src/WinUI/ArcGIS.WinUI.Viewer/Samples/Geometry/SnapGeometryEdits/SnapGeometryEdits.xaml.cs

```diff
@@ -0,0 +1,355 @@
+            foreach (var item in PointSnapSettingsList.Items.ToList())
+            {
+                if (item is SnapSourceSettingsVM snapSourceSettingsVM)
+                {
+                    snapSourceSettingsVM.IsEnabled = true;
+                }
+            }
+        }
+
+        // Enable all polyline layer snap sources.
+        private void EnableAllPolylineSnapSourceButton_Click(object sender, RoutedEventArgs e)
+        {
+            foreach (var item in PolylineSnapSettingsList.Items.ToList())
+            {
+                if (item is SnapSourceSettingsVM snapSourceSettingsVM)
+                {
+                    snapSourceSettingsVM.IsEnabled = true;
+                }
+            }
+        }
+        #endregion
+
+        #region Geometry Management Button Handlers
+        private void DeleteButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.DeleteSelectedElement();
+        }
+
+        private void UndoButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.Undo();
+        }
+
+        private void SaveButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_selectedGraphic?.Geometry != null)
+            {
+                _selectedGraphic.Geometry = _geometryEditor.Stop();
+                _selectedGraphic.IsSelected = false;
+            }
+            else if (_geometryEditor.IsStarted)
+            {
+                CreateNewGraphic();
+            }
+
+            ResetFromEditingSession();
+        }
+
+        private void DiscardButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.Stop();
+
+            ResetFromEditingSession();
+        }
+        #endregion
+
+        #region Geometry Tool Buttons Handlers
+        private void PointButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_geometryEditor.IsStarted)
+            {
+                _geometryEditor.Stop();
+            }
+
+            ResetFromEditingSession();
+
+            _geometryEditor.Start(GeometryType.Point);
+        }
+
+        private void MultipointButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_geometryEditor.IsStarted)
+            {
+                _geometryEditor.Stop();
+            }
+
+            ResetFromEditingSession();
+
+            _geometryEditor.Start(GeometryType.Multipoint);
+        }
+        private void PolylineButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_geometryEditor.IsStarted)
+            {
+                _geometryEditor.Stop();
+            }
+
+            ResetFromEditingSession();
+
+            _geometryEditor.Start(GeometryType.Polyline);
+        }
+        private void PolygonButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_geometryEditor.IsStarted)
+            {
+                _geometryEditor.Stop();
+            }
+
+            ResetFromEditingSession();
+
```

Comment:
Button changes color when pressed on MAUI sample to indicate that geometry editor is started with this geometry type. Can this be added to WinUI and WPF implementations?